commons/src/main/java/com/droidcon/commons/data/db/AppDatabase.kt

```diff
@@ -0,0 +1,39 @@
+package com.droidcon.commons.data.db
+
+import android.content.Context
+import androidx.room.Database
+import androidx.room.Room
+import androidx.room.RoomDatabase
+import com.droidcon.commons.data.db.entities.SessionEntity
+
+
+@Database(entities = [SessionEntity::class], version = 1)
+abstract class AppDatabase : RoomDatabase() {
+
+    abstract fun sessionDao(): SessionsDao
+
+    companion object {
```

Comment:
We can make the DB singleton by annotating it with `javax.inject.Singleton` annotation and providing the instance through dagger in the next way. It would like like the next:
```
@Provides
fun provideDatabase(appContext: Context): SessionizeDatabase = Room.databaseBuilder(
     appContext,
     SessionizeDatabase::class.java,
     SessionizeDatabase.NAME
).build()

@Provides
fun provideSpeakerDao(sessionizeDatabase: SessionizeDatabase): SpeakerDao =
    sessionizeDatabase.speakerDao()
```
WDYT about it?

Reply:
hmm is not very different from what we there is now, what advantages do you see for this approach over the other? With the actual approach building the database is syncronized in case multiple threads try to build it at the same time. Anyway I don't have a strong opinion on this.

Reply:
Only simplicity and consistency. 
Instance creation for singletons is also thread safe in Dagger, so we wouldn't need to write that code by ourselves.
Also moving the instantiation to the dagger module is better IMO. With the self implemented singleton we would also have to write a method in a dagger module to provide that concrete instance to classes using dagger to get their collaborators injected. Also, for all the classes that need to be instantiated in a special way (except view models) we're doing that in a method contained in a module, so it is also a matter of consistency. 
Anyways, this is really minor, both options are ok :)

Reply:
Agree then! did not know dagger singleton creation was thread safe for free. Good to know. Agree as well regarding consistency and simplicity. Will apply this changes asap